agent/agent_endpoint.go

```diff
@@ -288,43 +286,9 @@ func (s *HTTPServer) AgentService(resp http.ResponseWriter, req *http.Request) (
 				return "", nil, acl.ErrPermissionDenied
 			}
 
-			var connect *api.AgentServiceConnect
-			var proxy *api.AgentServiceConnectProxyConfig
-
-			if svc.Connect.Native {
-				connect = &api.AgentServiceConnect{
-					Native: svc.Connect.Native,
-				}
-			}
-
-			if svc.Kind == structs.ServiceKindConnectProxy ||
-				svc.Kind == structs.ServiceKindMeshGateway {
-
-				proxy = svc.Proxy.ToAPI()
-			}
-
-			var weights api.AgentWeights
-			if svc.Weights != nil {
-				err := mapstructure.Decode(svc.Weights, &weights)
-				if err != nil {
-					return "", nil, err
-				}
-			}
-
 			// Calculate the content hash over the response, minus the hash field
-			reply := &api.AgentService{
-				Kind:              api.ServiceKind(svc.Kind),
-				ID:                svc.ID,
-				Service:           svc.Service,
-				Tags:              svc.Tags,
-				Meta:              svc.Meta,
-				Port:              svc.Port,
-				Address:           svc.Address,
-				EnableTagOverride: svc.EnableTagOverride,
-				Weights:           weights,
-				Proxy:             proxy,
-				Connect:           connect,
-			}
+			aSvc := buildAgentService(svc)
```

Comment:
The only 3 things this function does differently are set `TaggedAddresses`, `CreateIndex`, and `ModifyIndex`. `TaggedAddresses` was the intended field for this patch. Will it be ok if the plumbing for the other two is involved?

Reply:
I think it should be okay. Unless someone using the API is explicitly expecting to not see the indexes. They seem to always be zero and excluded from the json anyways.

```
mkeeler-mbp:~ mkeeler$ curl localhost:8500/v1/agent/services
{
    "foo": {
        "ID": "foo",
        "Service": "foo",
        "Tags": [],
        "Meta": {},
        "Port": 5678,
        "Address": "",
        "TaggedAddresses": {
            "wan": {
                "Address": "198.18.0.1",
                "Port": 1234
            }
        },
        "Weights": {
            "Passing": 1,
            "Warning": 1
        },
        "EnableTagOverride": false
    }
}
mkeeler-mbp:~ mkeeler$ curl localhost:8500/v1/agent/service/foo
{
    "ID": "foo",
    "Service": "foo",
    "Tags": [],
    "Meta": {},
    "Port": 5678,
    "Address": "",
    "TaggedAddresses": {
        "wan": {
            "Address": "198.18.0.1",
            "Port": 1234
        }
    },
    "Weights": {
        "Passing": 1,
        "Warning": 1
    },
    "EnableTagOverride": false,
    "ContentHash": "4f25bb0e66da47d3"
}
mkeeler-mbp:~ mkeeler$
```

This actually brings up another point which is whether the `v1/agent/services` endpoint should compute the content hashes of the services before sending them back (or maybe this should be moved into `buildAgentService`